### PR TITLE
Create OverviewLayout component

### PR DESCRIPTION
**Pull request by sauldom102**
## Description

We could be more prescriptive with this layout as it's a common one that we'll use for all overview pages, not allowing to place anything else on the sidepanel apart from the `DetailsItemsList` component.

The idea with this component is to deprecate the `InfoPaneLayout` and stop exporting it from FactorialOne, this will close the door to some unexpected customization.

I wish we could have closed this even more, so that the `children` is not any html, but we have the Widgets abstraction in our main frontend codebase and we can't entirely do that.

Proposed here: https://factorialteam.slack.com/archives/C07LQSJ5UQY/p1731587350661809

## Screenshots

Screenshot 2024-11-14 at 14 02 47

### CODE REVIEW

**Review by CarlosBM18 — APPROVED** [review]
LGTM!

File: lib/experimental/PageLayouts/OverviewLayout/index.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const meta = {
+  component: OverviewLayout,
+  tags: ["autodocs"],
+  decorators: [PageDecorator],
+  args: {
+    children: <Placeholder className="h-[450px]">Main</Placeholder>,
+    sidepanel: {
+      title: DETAILS_ITEMS_ARGS.title,
+      items: DETAILS_ITEMS_ARGS.details,
+    },
```
Comment (josepjaume) [inline lib/experimental/PageLayouts/OverviewLayout/index.stories.tsx:24]: I'm worried that this won't scale well in the future, perhaps? But I guess it's easier to make an API less strict than more strict, so I'm not opposing the idea.

File: lib/experimental/PageLayouts/OverviewLayout/index.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  parameters: {
+    a11y: {
+      config: {
+        rules: [{ id: "svg-img-alt", enabled: false }],
+      },
```
Comment (josepjaume) [inline lib/experimental/PageLayouts/OverviewLayout/index.stories.tsx:30]: Do we need this?

File: lib/experimental/PageLayouts/OverviewLayout/index.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+export interface OverviewLayoutProps {
+  children: ReactNode
+  sidepanel: {
+    title: string
+    items: DetailsItemType[]
+  }
+}
+
+export const OverviewLayout = forwardRef<HTMLDivElement, OverviewLayoutProps>(
```
Comment (josepjaume) [inline lib/experimental/PageLayouts/OverviewLayout/index.tsx:14]: Way better name than `InfoPane`. Great job!

**Review by josepjaume — APPROVED** [review]
Great job! Please keep the design team on the loop. I think the naming makes total sense.

File: lib/experimental/PageLayouts/OverviewLayout/index.stories.tsx (hunk 2 above)
Comment (sauldom102) [reply lib/experimental/PageLayouts/OverviewLayout/index.stories.tsx:30]: Probably not, I just copied over some other story and forgot to remove that!

File: lib/experimental/PageLayouts/OverviewLayout/index.stories.tsx (hunk 1 above)
Comment (sauldom102) [reply lib/experimental/PageLayouts/OverviewLayout/index.stories.tsx:24]: Yep, maybe we end up having more cases, but I also thought it's easier to do that change in the future if we don't have this component too open and working in many different ways in our frontend

**Pull request merged**